app/models/wiki.rb

```diff
@@ -73,14 +74,18 @@ def api_url
   #############
 
   def ensure_valid_project
-    # Multilingual projects must have language == nil.
+    # Most multilingual projects must have language == nil,
+    # but wikisource is the exception, it can have nil, or a language.
     # Standard projects must have a language.
-    if MULTILINGUAL_PROJECTS.include?(project)
+    #
+    # TODO: Validate the language/project combination by pinging it's API.
+    if project == "wikidata"
       self.language = nil
+    elsif project == "wikisource"
```

Comment:
The setting of defaults is handled on the backend, so we currently pass in many nils from the frontend that get converted to English Wikipedia. See for example the `set_wiki` methods of CourseCreationManager and of AssignmentsController.

I think handling that "www" here will be the cleanest solution (although I could be convinced otherwise).